packages/eslint-plugin/src/rules/no-extra-parens.ts

```diff
@@ -123,6 +123,19 @@ export default util.createRule<Options, MessageIds>({
       },
       // AssignmentExpression
       // AwaitExpression
+      AwaitExpression(node) {
+        if (util.isTypeAssertion(node.argument)) {
+          // reduces the precedence of the node so the rule thinks it needs to be rapped
+          return rules.AwaitExpression({
+            ...node,
+            argument: {
+              ...node.argument,
+              type: AST_NODE_TYPES.SequenceExpression as any,
+            },
+          });
+        }
+        return rules.AwaitExpression(node);
```

Comment:
Indeed, looks like the case of an await expression _not_ on a type assertion isn't covered by tests. Could you add some in please?